app/services/email_service/ses/synchronize.rb

```diff
@@ -7,27 +7,30 @@ module EmailService::SES::Synchronize
   module_function
 
   def run_batch_synchronization!(ses_client:)
-    ses_client.list_verified_addresses.on_success do |vaddrs|
-      vaddrs = vaddrs.to_set
-      offset = 0
-      addresses = AddressStore.load_all(limit: BATCH_SIZE, offset: offset)
+    offset = 0
+    addresses = AddressStore.load_all(limit: BATCH_SIZE, offset: offset)
+
+    while addresses.present?
+      verified_addresses = []
+      addresses.each do |address|
+        if email_verified?(ses_client, address[:email])
+          verified_addresses.push(address[:email])
```

Comment:
In addition, `get_identity_verification_attributes` call will fail if an email in the list is not present in SES. This can happen, if something was manually deleted in SES. So to build the correct list of verified identities in SES, you need to load the identities from ses first with `list_identities` call (beware, that's paged at 1000, observe the `next_token` in response).